python/ray/serve/_private/router.py

```diff
@@ -74,8 +75,64 @@ async def resolve_async_tasks(self):
         scanner.clear()
 
 
-class ReplicaSet:
-    """Data structure representing a set of replica actor handles"""
+class ReplicaScheduler(ABC):
+    async def assign_replica(
+        self, query: Query
+    ) -> Union[ray.ObjectRef, ray._raylet.StreamingObjectRefGenerator]:
+        pass
+
+    def update_running_replicas(self, running_replicas: List[RunningReplicaInfo]):
+        pass
+
+
+class RoundRobinStreamingReplicaScheduler(ReplicaScheduler):
+    """Round-robins requests across a set of actor replicas using streaming calls.
+
+    This policy does *not* currently respect `max_concurrent_queries`.
+    """
+
+    def __init__(self):
+        self._replica_iterator = itertools.cycle([])
+        self._replicas_updated_event = asyncio.Event()
+
+    async def assign_replica(
+        self, query: Query
+    ) -> ray._raylet.StreamingObjectRefGenerator:
+        replica = None
+        while replica is None:
+            try:
+                replica = next(self._replica_iterator)
+            except StopIteration:
```

Comment:
what is the scenario for replica not available?

Looks like we never count any ongoing requests.

Reply:
In the case there are no replicas available this loop will repeat forever